src/PruneUnusedImagesModal.jsx

```diff
@@ -49,81 +49,71 @@ function ImageOptions({ images, checked, isSystem, handleChange, name, showCheck
-        const showCheckboxes = userImages.length > 0 && systemImages.length > 0;
-        return (
-            <Modal isOpen
-                   onClose={this.props.close}
-                   position="top" variant="medium"
-                   title={cockpit.format(_("Prune unused images"))}
-                   footer={<>
-                       <Button id="btn-img-delete" variant="danger"
-                               spinnerAriaValueText={this.state.isPruning ? _("Pruning images") : undefined}
-                               isLoading={this.state.isPruning}
-                               isDisabled={!this.state.deleteUserImages && !this.state.deleteSystemImages}
-                               onClick={this.handlePruneUnusedImages}>
-                           {this.state.isPruning ? _("Pruning images") : _("Prune")}
-                       </Button>
-                       <Button variant="link" onClick={() => this.props.close()}>{_("Cancel")}</Button>
-                   </>}
-            >
-                <Flex flex={{ default: 'column' }}>
-                    {isSystem && <ImageOptions
-                  images={systemImages}
-                  name="deleteSystemImages"
-                  checked={this.state.deleteSystemImages}
-                  handleChange={this.handleChange}
-                  showCheckbox={showCheckboxes}
-                  isSystem
-                    />
-                    }
-                    <ImageOptions
-                  images={userImages}
-                  name="deleteUserImages"
-                  checked={this.state.deleteUserImages}
-                  handleChange={this.handleChange}
-                  showCheckbox={showCheckboxes}
-                  isSystem={false}
-                    />
-                </Flex>
-            </Modal>
-        );
-    }
-}
+    return (
+        <Modal isOpen
+               onClose={close}
+               position="top" variant="medium"
+               title={cockpit.format(_("Prune unused images"))}
+               footer={<>
+                   <Button id="btn-img-delete" variant="danger"
+                           spinnerAriaValueText={isPruning ? _("Pruning images") : undefined}
+                           isLoading={isPruning}
+                           isDisabled={!deleteUserImages && !deleteSystemImages}
+                           onClick={handlePruneUnusedImages}>
+                       {isPruning ? _("Pruning images") : _("Prune")}
+                   </Button>
+                   <Button variant="link" onClick={() => close()}>{_("Cancel")}</Button>
+               </>}
+        >
+            <Flex flex={{ default: 'column' }}>
+                {isSystem && <ImageOptions
+              images={systemImages}
+              name="deleteSystemImages"
+              checked={deleteSystemImages}
+              handleChange={(checked, _) => setDeleteSystemImages(checked)}
```

Comment:
Same here, let's use the single function name way with implicit parameter passing if you agree.